cmd/soroban-rpc/internal/daemon/daemon.go

```diff
@@ -143,19 +180,27 @@ func MustNew(cfg config.LocalConfig) *Daemon {
 		}
 	}
 
-	ingestService, err := ingest.NewService(ingest.Config{
-		Logger:            logger,
-		DB:                db.NewReadWriter(dbConn, maxLedgerEntryWriteBatchSize, maxRetentionWindow),
-		EventStore:        eventStore,
-		TransactionStore:  transactionStore,
-		NetworkPassPhrase: cfg.NetworkPassphrase,
-		Archive:           historyArchive,
-		LedgerBackend:     core,
-		Timeout:           cfg.LedgerEntryStorageTimeout,
-	})
+	ingestService, err := backoff.RetryNotifyWithData(
+		func() (*ingest.Service, error) {
+			return ingest.NewService(ingest.Config{
```

Comment:
Good catch. I changed the signature so that it no longer returns an error, and added constant backoff (retry every second for 5 seconds) to running ingestion.